### PR TITLE
Address issues picked up by clazy/clang-tidy

### PR DESCRIPTION
**Pull request by piggz**

### CODE REVIEW

**Comment by rinigus** [comment]
There are a lot of autogenerated files submitted as this PR. Please edit the commit (reset HEAD~) and add only source files.

File: src/chum.cpp [hunk 1]
```diff
@@ -31,7 +32,7 @@ Chum::Chum(QObject *parent)
 {
   connect(&m_ssu, &Ssu::updated, this, &Chum::repositoriesListUpdated);
   connect(&m_ssu, &Ssu::updated, this, &Chum::repoUpdated);
-  connect(Daemon::global(), &Daemon::updatesChanged, [this]() { this->getUpdates(); });
+  connect(Daemon::global(), &Daemon::updatesChanged, this, [this]() { this->getUpdates(); });
```
Comment (rinigus) [inline src/chum.cpp:36]: As we have lambda functions, I think that it is way clearer to use the original syntax and not add `this` to `connect`.

File: src/chum.cpp [hunk 2]
```diff
@@ -124,7 +125,7 @@ void Chum::refreshPackagesInstalled()
 {
   // check what repository provides installed packages
   QHash<QString,QString> packages;
-  for (const auto &p: m_packages_last_refresh_installed)
+  for (const auto &p: std::as_const(m_packages_last_refresh_installed))
```
Comment (rinigus) [inline src/chum.cpp:128]: according to https://en.cppreference.com/w/cpp/utility/as_const, it is from C++17. maybe an issue for SFOS 3.1. In this context, I don't see much benefit of having it either

File: src/chum.cpp [hunk 3]
```diff
@@ -167,7 +168,7 @@ void Chum::refreshPackagesFinished()
     package->setPkidLatest(p);
   }
 
-  setStatus(QStringLiteral());
+  setStatus(QLatin1String());
```
Comment (rinigus) [inline src/chum.cpp:171]: I wouldn't replace it here. Interesting post in this regard: https://forum.qt.io/topic/78540/qstringliteral-vs-qlatin1string

As we set the value to m_status - which is a qstring, we may create empty qstring straight away and avoid conversion.

File: src/chum.cpp [hunk 4]
```diff
@@ -181,7 +182,7 @@ void Chum::refreshDetails() {
   setStatus(qtTrId("chum-get-package-details"));
 
   QStringList packages;
-  for (const ChumPackage *p: m_packages.values())
+  for (const ChumPackage *p: std::as_const(m_packages))
```
Comment (rinigus) [inline src/chum.cpp:185]: `m_packages` is a QHash and I am surprised that such iterator would work. Shouldn't it iterate via pairs of key,value?  As there are many extra files in this PR, I didn't check whether it compiles...

For clarity of reading, I would prefer older version where it is clear that iteration is done through values

**Review by rinigus — CHANGES_REQUESTED** [review]
All in all, I would drop many of suggested changes (see comments). Some later changes in header and simpler iterators (where const was forgotten) make sense and could be applied.

File: src/chum.cpp (hunk 4 above)
Comment (piggz) [reply src/chum.cpp:185]: Yes it does, and is the recommended way of iterating the hash values (not the keys) See https://doc.qt.io/qt-5/qhash.html#details

File: src/chum.cpp (hunk 2 above)
Comment (piggz) [reply src/chum.cpp:128]: Ok... qt uses qAsConst() macro for the same thing, but this was added in 5.7.

File: src/chum.cpp (hunk 1 above)
Comment (piggz) [reply src/chum.cpp:36]: The check is this one https://github.com/KDE/clazy/blob/master/docs/checks/README-connect-3arg-lambda.md.  The this object is the context parameter which ensures the lambda isnt executed if the object is deleted.  Im no expert, and merely relying on the clazy devs knowledge here.

Comment (rinigus) [reply src/chum.cpp:36]: Interesting point, then it does make sense

File: src/chum.cpp (hunk 2 above)
Comment (rinigus) [reply src/chum.cpp:128]: I would suggest to skip it as we need to support older SFOS versions

File: src/chum.cpp (hunk 4 above)
Comment (rinigus) [reply src/chum.cpp:185]: Really odd, as I see example

```C++
QHash<QString, int>::const_iterator i = hash.constBegin();
while (i != hash.constEnd()) {
    cout << i.key() << ": " << i.value() << Qt::endl;
    ++i;
}
```

which means that you cannot assign iterator to `const ChumPackage *p`, but have to get the package via `i.value()`. Something is odd over here, I would have to try to compile whole thing

Comment (piggz) [reply src/chum.cpp:185]: Its the last example:
If you only need to extract the values from a hash (not the keys), you can also use foreach:

QHash<QString, int> hash;
...
foreach (int value, hash)
    cout << value << Qt::endl;

File: src/chum.cpp (hunk 2 above)
Comment (piggz) [reply src/chum.cpp:128]: For future reference, ill link the check https://github.com/KDE/clazy/blob/master/docs/checks/README-range-loop-detach.md

Comment (rinigus) [reply src/chum.cpp:128]: See failures at https://build.sailfishos.org/package/show/sailfishos:chum:testing/sailfishos-chum-gui . I have set revision to this branch commit

File: src/chum.cpp (hunk 4 above)
Comment (rinigus) [reply src/chum.cpp:185]: But we don't use `foreach`. I was sure that such `for` loop form corresponds to iteration using begin() and end(). So, it is rather confusing, I must say

Comment (rinigus) [reply src/chum.cpp:185]: Experimented a bit and, indeed, such iteration goes through values and not the iterator. I guess we can then just make this change. Although, without as_const

**Review by rinigus — APPROVED** [review]

**Comment by rinigus** [comment]
@piggz: looks good by me, now have to wait for OBS build results. Would you want to supplement this with one commit with reformatting whole source?

**Pull request merged**